src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/CpxVariantDetector.java

```diff
@@ -0,0 +1,782 @@
+            } else {
+                // depending on the representation and the current alignment's orientation, traverse segments in different order
+                final int start, stop, step;
+                if (basicInfo.forwardStrandRep == alignment.forwardStrand) {
+                    start = 0;
+                    stop = segments.size();
+                    step = 1;
+                } else {
+                    start = segments.size()-1;
+                    stop = -1;
+                    step = -1;
+                }
+                // N*M overlaps
+                int i = start;
+                for ( ; i != stop; i += step) {
+                    final SimpleInterval currentSegment = segments.get(i);
+                    // if current segment is contained in current alignment, note it down
+                    if (alignmentContainsSegment(alignment.referenceSpan, currentSegment)) {
+                        if (basicInfo.forwardStrandRep) // +1 below on i for 1-based description, no magic
+                            descriptions.add( String.valueOf((alignment.forwardStrand ? 1 : -1) * (i+1)) );
+                        else
+                            descriptions.add( String.valueOf((alignment.forwardStrand ? -1 : 1) * (i+1)) );
+                    }
+
+                }
+                // if the current alignment is associated with a gapped jump,
+                // we need to signal that an unmapped insertion is present,
+                // but only under two cases:
+                //  1) no more segments to explore
+                //  2) the next segment IS NOT contained in the current alignment's ref span
+                if ( currentJumpGapSize > 0 && !jumpIsLast){
+                    descriptions.add(ReferenceSegmentsAndEventDescription.UNMAPPED_INSERTION + "-" + currentJumpGapSize);
+                }
+            }
+
+            if (jumpGapSizeIterator.hasNext()) // last alignment has no leaving jump so need to guard against that
+                currentJumpGapSize = jumpGapSizeIterator.next();
+            else
+                jumpIsLast = true;
+        }
+
+        // post-hoc treatment of insertions that map to disjoint regions
+        // go in reverse order because inserting into "descriptions", and insertion invalidates indices/iterators
+        final ImmutableList<Tuple2<SimpleInterval, Integer>> reverse =
+                ImmutableList.copyOf(insertionMappedToDisjointRegionAndWhereToInsert).reverse();
+        for (final Tuple2<SimpleInterval, Integer> pair : reverse){
+            final int index = pair._2;
+            if (index > 0) {
+                descriptions.add(pair._2, pair._1.toString());
+            } else {
+                descriptions.add(-1*pair._2, "-"+pair._1.toString());
+            }
+        }
+        return descriptions;
+    }
+
+    // avoid case where segment and alignment ref span overlap only on one boundary base
```

Comment:
What's the case where `SimpleInterval.contains()` doesn't work for this? Should we fix that method?

Reply:
that's right, and now replaced with `SimpleInterval.contains()`